mesonbuild/coredata.py

```diff
@@ -40,13 +40,50 @@
 import typing as T
 
 if T.TYPE_CHECKING:
+    from typing_extensions import Protocol
+
     from . import dependencies
     from .compilers.compilers import Compiler, CompileResult, RunResult, CompileCheckMode
     from .dependencies.detect import TV_DepID
     from .environment import Environment
     from .mesonlib import FileOrString
     from .cmake.traceparser import CMakeCacheEntry
 
+    class CommandLineOptions(Protocol):
+
+        """Representation of command line options from Meson setup and Meson
+        configure.
+
+        :param builddir: The path to the build directory
+        :param clearcache: whether to wipe caches
+        :param cmd_line_options: Raw options converted into `OptionKey: str`
+            pairs
+        :param cross_file: List of cross files passed
+        :param fatal_warnings: whether Meson warnings should cause configuration
+            to stop
+        :param native_file: List of native files pass
+        :param pager: Should Meson start a pager for it's configure output?
+        :param profile: whether to profile the meson configuration step
+        :param project_options: A list of row option passed on the command line
+        :param reconfigure: should Meson should force a fall reconfigure even if
+            it's otherwise unnecessary
+        :param sourcedir: The path to the source directory
+        :param wipe: whether to wipe the build dir
```

Comment:
This is really just duplicating the argparse help, we don't do that for the other argparse Protocols we define elsewhere.

Reply:
Okay, in the new version I've removed all of the param docs except those that don't match the command line arguments, and noted that the rest are just the arguments from the command line.